### ISSUE TITLE
Improve angular signals benchmark performance

### ISSUE THREAD

**Issue by NateRadebaugh** [angular/angular#58846]
### Description

According to public benchmarks, angular's signals are much worse than alternative ecosystem implementations.

> https://github.com/transitive-bullshit/js-reactivity-benchmark

This makes angular look inferior from an outsider's viewpoint.

### Proposed solution

Unknown 

### Alternatives considered

Unknown

**Comment by JoostK** [comment]
We're aware; Angular's signals implementation only using an eager dirty propagation model when evaluated within an effect. Outside of effects, downstream computeds have to check their sources on every read to see if they need to reevaluate. This model has benefits for signal lifetime management. I expect this to be the primary contributor to the benchmark results.

**Comment by sod** [comment]
We use observable everywhere as we run zoneless + OnPush and rely on rxjs to notify angular where an when to update. Does this benchmark result mean, that our app would get slower if we were to migrate from rxjs to signals? Or does rxjs already has the same performance hit?

**Comment by pkozlowski-opensource** [comment]
Note: as with any low-level benchmark, this one is mostly useful for people working on the internals of the signals library. As @JoostK mentioned, the mentioned benchmark exercises many code paths and scenarios that are not really relevant to Angular applications using signals.

tl;dr; that this benchmark should not be a decision factor when it comes to using / not using signals.

**Comment by PowerKiKi** [comment]
More information on https://x.com/synalx/status/1868281397179277382 where it is also confirmed that, in real-world applications, Angular signals speed is close to Vue's.

**Comment by milomg** [comment]
Update: Angular 20's `Injector.destroy` enables Angular signals to be pretty competitive on the latest update of the js-reactivity-benchmark

https://github.com/milomg/js-reactivity-benchmark

**Comment by JeanMeche** [comment]
Thanks for updating your benchmark @milomg 

There are 2 key informations that I want to highlight before closing this issue. 

* Benchmark are only as good as to what they are actually benchmarking. Sometimes slight details can have huge impact pure on numbers than won't reflect in real life applications
* Will continue improving the performance of Angular signatures if we discover important new performance bottlenecks.
